### PR TITLE
Added edit_area() function

### PR DESCRIPTION
**Pull request by Prathamesh-Ghatole**
Based on #4

Adds an "edit_area()" function to editing.py for editing existing areas in MusicBrainz using existing and updated information.
Includes tests for the edit_area() function

Updates the test_add_area.py test script to follow best practices.

### CODE REVIEW

File: tests/test_edit_area.py
Comment (reosarevok) [inline tests/test_edit_area.py]: To edit, not add :)

File: musicbrainz_bot/editing.py
Comment (reosarevok) [inline musicbrainz_bot/editing.py:329]: I might be missing something, but what's the benefit of passing edit note separately and then adding it to `update` here? Is it to keep the dict structure?

File: tests/test_edit_area.py
Comment (reosarevok) [inline tests/test_edit_area.py]: This does not test any of the values were updated, does it?

**Review by reosarevok — CHANGES_REQUESTED** [review]

File: tests/test_edit_area.py
Comment (Prathamesh-Ghatole) [reply tests/test_edit_area.py]: 🤦

File: tests/test_edit_area.py
Comment (Prathamesh-Ghatole) [reply tests/test_edit_area.py]: On it :)

Comment (Prathamesh-Ghatole) [reply tests/test_edit_area.py]: Done! Except for a tiny issue, that I am writing in another comment on this PR thread.

File: musicbrainz_bot/editing.py
Comment (Prathamesh-Ghatole) [reply musicbrainz_bot/editing.py:329]: Well, that's because the in every other function in this bot, the edit note is generally passed as a parameter to the edit function, and not in the data dictionary itself.

I thought following the convention might be a good idea :))

Comment (Prathamesh-Ghatole) [reply musicbrainz_bot/editing.py:329]: We can easily turn the function around to receive the edit note in the dict itself if you wish.

File: musicbrainz_bot/editing.py
Comment (Prathamesh-Ghatole) [inline musicbrainz_bot/editing.py:328]: @reosarevok Now that the newly edited values can be cross-checked (see tests/test_edit_area.py), the only problem is the fact that my function can't REMOVE existing URLs. It can only add them. 

That's because removing existing URLs requires querying the database or something for the URL ID, which is automatically done on the musicbrainz.org frontend (somehow), but manually submitting requests also requires the user to specify the ID of the URL to be removed in the request itself, which I have no idea how to access.

Any idea how I could get around doing that?

File: tests/test_add_area.py
Comment (reosarevok) [inline tests/test_add_area.py]: FWIW I don't think we need the comments, they mostly reword the test failure label.

File: musicbrainz_bot/editing.py
Comment (reosarevok) [reply musicbrainz_bot/editing.py:329]: Ok, I guess it's true it's separate elsewhere :)

File: tests/test_add_area.py
Comment (Prathamesh-Ghatole) [reply tests/test_add_area.py]: On it :)

File: musicbrainz_bot/editing.py
Comment (Prathamesh-Ghatole) [reply musicbrainz_bot/editing.py:329]: Soo should we keep it separate here as well, or would you like me to merge it?

File: musicbrainz_bot/editing.py
Comment (reosarevok) [inline musicbrainz_bot/editing.py]: How is it going with this? :)

Comment (mwiencek) [reply musicbrainz_bot/editing.py]: I think it's fine to merge this PR if it's working for *adding* URLs, for now, but then I'd remove this WIP code and put it in another branch to propose later once it's working.

File: tests/test_edit_area.py
Comment (mwiencek) [inline tests/test_edit_area.py]: It'd be nice to have the URL tests working before merging. Was there an issue here?

File: musicbrainz_bot/editing.py
Comment (Prathamesh-Ghatole) [reply musicbrainz_bot/editing.py]: Yes, currently URLs can only be added. Not Removed. I can remove the WIP code rn if you want.

File: tests/test_edit_area.py
Comment (Prathamesh-Ghatole) [reply tests/test_edit_area.py]: That's because removing existing URLs requires querying the database or something for the URL ID, which is automatically done on the musicbrainz.org frontend (somehow), but manually submitting requests also requires the user to specify the ID of the URL to be removed in the request itself, which I have no idea how to access.

Any idea how I could get around doing that?

Comment (mwiencek) [reply tests/test_edit_area.py]: Yeah, that's likely where we'll need to make use of the direct PG connection to the database. `SELECT id FROM url WHERE url = ?`.

Comment (Prathamesh-Ghatole) [reply tests/test_edit_area.py]: Alright, I'll give it a shot ASAP

File: musicbrainz_bot/editing.py
Comment (mwiencek) [reply musicbrainz_bot/editing.py]: It would help keep the git history clean. But we can do it before merging.

**Pull request merged**
